monai/utils/type_conversion.py

```diff
@@ -31,36 +32,29 @@
     "convert_to_cupy",
     "convert_to_numpy",
     "convert_to_tensor",
+    "convert_to_meta_tensor",
     "convert_to_dst_type",
 ]
 
 
-_torch_to_np_dtype = {
-    torch.bool: np.dtype(bool),
-    torch.uint8: np.dtype(np.uint8),
-    torch.int8: np.dtype(np.int8),
-    torch.int16: np.dtype(np.int16),
-    torch.int32: np.dtype(np.int32),
-    torch.int64: np.dtype(np.int64),
-    torch.float16: np.dtype(np.float16),
-    torch.float32: np.dtype(np.float32),
-    torch.float64: np.dtype(np.float64),
-    torch.complex64: np.dtype(np.complex64),
-    torch.complex128: np.dtype(np.complex128),
-}
-_np_to_torch_dtype = {value: key for key, value in _torch_to_np_dtype.items()}
+def get_numpy_dtype_from_string(dtype: str) -> np.dtype:
+    """Get a numpy dtype (e.g., `np.float32`) from its string (e.g., `"float32"`)."""
+    return np.zeros([], dtype=dtype).dtype  # type: ignore
 
 
-def dtype_torch_to_numpy(dtype):
+def get_torch_dtype_from_string(dtype: str) -> torch.dtype:
+    """Get a torch dtype (e.g., `torch.float32`) from its string (e.g., `"float32"`)."""
+    return dtype_numpy_to_torch(get_numpy_dtype_from_string(dtype))
+
+
+def dtype_torch_to_numpy(dtype: torch.dtype) -> np.dtype:
     """Convert a torch dtype to its numpy equivalent."""
-    return look_up_option(dtype, _torch_to_np_dtype)
+    return torch.zeros([], dtype=dtype).numpy().dtype  # type: ignore
 
 
-def dtype_numpy_to_torch(dtype):
+def dtype_numpy_to_torch(dtype: np.dtype) -> torch.dtype:
     """Convert a numpy dtype to its torch equivalent."""
-    # np dtypes can be given as np.float32 and np.dtype(np.float32) so unify them
-    dtype = np.dtype(dtype) if isinstance(dtype, (type, str)) else dtype
-    return look_up_option(dtype, _np_to_torch_dtype)
+    return torch.from_numpy(np.zeros([], dtype=dtype)).dtype
 
 
 def get_equivalent_dtype(dtype, data_type):
```

Comment:
I was looking at this, shall we make the `dtype` of `get_equivalent_dtype` support string input such as `"torch.float32"`

Reply:
sure.

Reply:
thanks! fyi... the string dtype is used here to store the type in applied operations (the np.dtype/torch.dtype object is not compatible with the collate function) and will need a revision after this PR https://github.com/Project-MONAI/MONAI/blob/fc5c34065326342ad91e08dea475614e16c42154/monai/transforms/spatial/array.py#L496